mozilla/fx-private-relay
# fix mpp #1270: lower() all subdomains

This also includes a 1-time migration to lower-case all existing
subdomains and to clean out any accidental case-sensitive duplicate
subdomains that may have been added already.

## Test

1. On `main` branch, make 2 premium accounts: A and B
2. With account A, register a subdomain with capital letters, e.g., `DoGHouSe`
3. With account B, register the same subdomain with different case letters, e.g., `dOghOUsE`
   * Expected: on `main` this will be allowed 😢 
4. Switch to this branch `fix-case-insensitive-domains-mpp-1270`
5. To clear out account B's erroneously subdomain, run `python manage.py migrate emails`
6. With account B, go to http://127.0.0.1:8000/accounts/profile/
   * Expected: account B's subdomain should now be empty again
7. Try to register the different-cased subdomain again, e.g., `dOghOUsE`
   * Expected: error "The domain @⁨dOghOUsE is not available. Please try again with another domain."
   * TODO: should we make the error show lower-case
8. Sign out of account B
9. Sign into account A
10. Should see domain as `doghouse` instead of `DoGHouSe`
